File: solutions/c/reverse-string/1/reverse_string.c
#include "reverse_string.h"

#include <stdlib.h>

char *reverse(const char *value)
{
    int i = 0;
    while (value[i])
        i++;
    int l = i - 1;
    char *output = (char *)malloc((l + 2) * sizeof(char));
    while (i >= 0)
    {
        i--;
        output[l - i] = value[i];
    }
    output[l + 1] = '\0';
return (output);
}

File: solutions/javascript/reverse-string/1/reverse-string.js
//
// This is only a SKELETON file for the 'Reverse String' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const reverseString = word => {
  let reversedWord = "";
  for (let i = 0; i < word.length; i++) {
    reversedWord = word[i] + reversedWord;
  }
  return reversedWord;
};
